gflownet/envs/lattice_parameters.py

```diff
@@ -0,0 +1,446 @@
+            min_step_len=min_step_len,
+            max_step_len=max_step_len,
+            **kwargs,
+        )
+
+        if lattice_system not in LATTICE_SYSTEMS:
+            raise ValueError(
+                f"Expected one of the keys or values from {LATTICE_SYSTEMS}, received {lattice_system}."
+            )
+
+        self.lattice_system = lattice_system
+        self.min_length = min_length
+        self.max_length = max_length
+        self.min_angle = min_angle
+        self.max_angle = max_angle
+        self.grid_size = grid_size
+
+        self.cell2angle = {
+            k: v for k, v in enumerate(np.linspace(min_angle, max_angle, grid_size))
+        }
+        self.angle2cell = {v: k for k, v in self.cell2angle.items()}
+        self.cell2length = {
+            k: v for k, v in enumerate(np.linspace(min_length, max_length, grid_size))
+        }
+        self.length2cell = {v: k for k, v in self.cell2length.items()}
+
+        if (
+            90.0 not in self.cell2angle.values()
+            or 120.0 not in self.cell2angle.values()
+        ):
+            raise ValueError(
+                f"Given min_angle = {min_angle}, max_angle = {max_angle} and grid_size = {grid_size}, "
+                f"possible discrete angle values {tuple(self.cell2angle.values())} do not include either "
+                f"the 90 degrees or 120 degrees angle, which must both be present."
+            )
+
+        self._set_source()
+        self.reset()
+
+    def _set_source(self):
+        """
+        Helper that sets self.source depending on the given self.lattice_system. For systems that have
+        specific angle requirements, they will be preset to these values.
+        """
+        if self.lattice_system == CUBIC:
+            angles = [90.0, 90.0, 90.0]
+        elif self.lattice_system == HEXAGONAL:
+            angles = [90.0, 90.0, 120.0]
+        elif self.lattice_system == MONOCLINIC:
+            angles = [90.0, self.min_angle, 90.0]
+        elif self.lattice_system == ORTHORHOMBIC:
+            angles = [90.0, 90.0, 90.0]
+        elif self.lattice_system == RHOMBOHEDRAL:
+            angles = [self.min_angle, self.min_angle, self.min_angle]
+        elif self.lattice_system == TETRAGONAL:
+            angles = [90.0, 90.0, 90.0]
+        elif self.lattice_system == TRICLINIC:
+            angles = [self.min_angle, self.min_angle, self.min_angle]
+        else:
+            raise NotImplementedError(
+                f"Unspecified lattice system {self.lattice_system}."
+            )
+
+        self.source = [0, 0, 0] + [self.angle2cell[angle] for angle in angles]
+
+    def get_actions_space(self) -> List[Tuple[int]]:
+        """
+        Constructs list with all possible actions, including eos.
+
+        The action is described by a tuple of dimensions (possibly duplicate) that will all be incremented
+        by 1, e.g. (0, 0, 0, 2, 4, 4, 4) would increment the 0th and the 4th dimension by 3, and 2nd by 1.
```

Comment:
Well, the first part of the description is correct, right? Action is a tuple of dimensions, and for each d in action, we increment `state[d]` by 1? So with `(0, 0, 0, 2, 4, 4, 4)`, we increment `state[0]` by 1 three times, `state[2]` one time, and `state[4]` three times? Because now I am confused if I got something wrong.